src/libmongoc/tests/json-test-operations.c

```diff
@@ -2260,37 +2260,46 @@ assert_session_transaction_state (json_test_ctx_t *ctx, const bson_t *operation)
 {
    const char *expected_state;
    const mongoc_client_session_t *session;
+   mongoc_transaction_state_t state;
+   char *state_strings[] = {"none",
+                            "starting",
+                            "in progress",
+                            "ending",
+                            "committed",
+                            "committed empty",
+                            "aborted"};
 
    expected_state = bson_lookup_utf8 (operation, "arguments.state");
    session = session_from_name (
       ctx, bson_lookup_utf8 (operation, "arguments.session"));
+   state = mongoc_client_session_get_transaction_state (session);
 
    if (!strcmp (expected_state, "none")) {
-      if (session->txn.state != MONGOC_INTERNAL_TRANSACTION_NONE) {
-         test_error ("expected session transaction state none, but have %i",
-                     session->txn.state);
+      if (state != MONGOC_INTERNAL_TRANSACTION_NONE) {
```

Comment:
lastly `MONGOC_INTERNAL_TRANSACTION_NONE` should be `MONGOC_TRANSACTION_NONE` to correspond to the states in `mongoc_transaction_state_t` as opposed to `mongoc_internal_transaction_state_t`

Reply:
Done!